pkg/config/auth.go

```diff
@@ -0,0 +1,1124 @@
+		a.Hook.SendEmail.Secrets = *remoteConfig.HookSendEmailSecrets
+	}
+
+	// Send SMS
+	if remoteConfig.HookSendSmsEnabled != nil {
+		a.Hook.SendSMS.Enabled = *remoteConfig.HookSendSmsEnabled
+	}
+	if remoteConfig.HookSendSmsUri != nil {
+		a.Hook.SendSMS.URI = *remoteConfig.HookSendSmsUri
+	}
+	if remoteConfig.HookSendSmsSecrets != nil {
+		a.Hook.SendSMS.Secrets = *remoteConfig.HookSendSmsSecrets
+	}
+}
+
+func (a *auth) mapRemoteSmsProviders(remoteConfig v1API.AuthConfigResponse) {
+	// Twilio
+	if remoteConfig.SmsTwilioAccountSid != nil && remoteConfig.SmsTwilioAuthToken != nil {
+		a.Sms.Twilio.Enabled = true
+		a.Sms.Twilio.AccountSid = *remoteConfig.SmsTwilioAccountSid
+		a.Sms.Twilio.AuthToken = *remoteConfig.SmsTwilioAuthToken
+		if remoteConfig.SmsTwilioMessageServiceSid != nil {
+			a.Sms.Twilio.MessageServiceSid = *remoteConfig.SmsTwilioMessageServiceSid
+		}
+	}
+
+	// Twilio Verify
+	if remoteConfig.SmsTwilioVerifyAccountSid != nil && remoteConfig.SmsTwilioVerifyAuthToken != nil {
+		a.Sms.TwilioVerify.Enabled = true
+		a.Sms.TwilioVerify.AccountSid = *remoteConfig.SmsTwilioVerifyAccountSid
+		a.Sms.TwilioVerify.AuthToken = *remoteConfig.SmsTwilioVerifyAuthToken
+		if remoteConfig.SmsTwilioVerifyMessageServiceSid != nil {
+			a.Sms.TwilioVerify.MessageServiceSid = *remoteConfig.SmsTwilioVerifyMessageServiceSid
+		}
+	}
+
+	// Messagebird
+	if remoteConfig.SmsMessagebirdAccessKey != nil {
+		a.Sms.Messagebird.Enabled = true
+		a.Sms.Messagebird.AccessKey = *remoteConfig.SmsMessagebirdAccessKey
+		if remoteConfig.SmsMessagebirdOriginator != nil {
+			a.Sms.Messagebird.Originator = *remoteConfig.SmsMessagebirdOriginator
+		}
+	}
+
+	// Textlocal
+	if remoteConfig.SmsTextlocalApiKey != nil {
+		a.Sms.Textlocal.Enabled = true
+		a.Sms.Textlocal.ApiKey = *remoteConfig.SmsTextlocalApiKey
+		if remoteConfig.SmsTextlocalSender != nil {
+			a.Sms.Textlocal.Sender = *remoteConfig.SmsTextlocalSender
+		}
+	}
+
+	// Vonage
+	if remoteConfig.SmsVonageApiKey != nil && remoteConfig.SmsVonageApiSecret != nil {
+		a.Sms.Vonage.Enabled = true
+		a.Sms.Vonage.ApiKey = *remoteConfig.SmsVonageApiKey
+		a.Sms.Vonage.ApiSecret = *remoteConfig.SmsVonageApiSecret
+		if remoteConfig.SmsVonageFrom != nil {
+			a.Sms.Vonage.From = *remoteConfig.SmsVonageFrom
+		}
+	}
+}
+
+func (original *auth) compareAndHideSensitiveFields(remote *auth) {
+	// This function compares the original auth struct with a remote auth struct
+	// and hides sensitive fields in both structs for secure comparison
+	// SMTP sensitive fields
+	diff.CompareSensitiveField(&original.Email.Smtp.AdminEmail, &remote.Email.Smtp.AdminEmail)
+	diff.CompareSensitiveField(&original.Email.Smtp.Host, &remote.Email.Smtp.Host)
+	diff.CompareSensitiveField(&original.Email.Smtp.User, &remote.Email.Smtp.User)
+	diff.CompareSensitiveField(&original.Email.Smtp.SenderName, &remote.Email.Smtp.SenderName)
+	diff.CompareSensitiveField(&original.Email.Smtp.Pass, &remote.Email.Smtp.Pass)
```

Comment:
We probably only need to treat secrets, api keys, and passwords as sensitive. Other strings are ok